module/VuFindSearch/src/VuFindSearch/Backend/Solr/Connector.php

```diff
@@ -332,6 +297,33 @@ public function query($handler, ParamBag $params, bool $cacheable = false)
         return $this->trySolrUrls($method, $urlSuffix, $callback, $cacheable);
     }
 
+    /**
+     * Call a method with provided options for the HTTP client
+     *
+     * @param array  $options HTTP client options
+     * @param string $method  Method to call
+     * @param array  ...$args Method parameters
+     *
+     * @return mixed
+     */
+    public function callWithHttpOptions(
```

Comment:
Thanks for all the progress here; I think this is coming along nicely. Just one last question: do we really want to allow arbitrary methods to be called via this public method? As written, you could easily call protected or private methods from the outside if you wanted to by using this proxy method... which I don't think is the intended effect. Do we either want to use reflection to only permit calls to public methods, or to create a fixed list of legal methods, just to be on the safe side? Or do we want to make this method protected, and then wrap it with a public `writeWithHttpOptions` method?

Reply:
Good point. I added a reflection-based check and tests.